src/main/java/gyro/google/compute/ImageFinder.java

```diff
@@ -0,0 +1,129 @@
+ * -------
+ *
+ * .. code-block:: gyro
+ *
+ *    compute-image: $(external-query google::compute-image { name: 'image-example' })
+ *
+ * .. code-block:: gyro
+ *
+ *    compute-image: $(external-query google::compute-image { family: 'image-family-example' })
+ *
+ * .. code-block:: gyro
+ *
+ *    compute-image: $(external-query google::compute-image { project: 'centos-cloud', family: 'centos-6' })
+ */
+@Type("compute-image")
+public class ImageFinder extends GoogleFinder<Compute, Image, ImageResource> {
+
+    private String name;
+    private String family;
+    private String project;
+
+    /**
+     * The name of the image.
+     */
+    public String getName() {
+        return name;
+    }
+
+    public void setName(String name) {
+        this.name = name;
+    }
+
+    /**
+     * The family of the image.
+     */
+    public String getFamily() {
+        return family;
+    }
+
+    public void setFamily(String family) {
+        this.family = family;
+    }
+
+    /**
+     * The project where the image resides. Defaults to current project.
+     */
+    public String getProject() {
+        return project;
+    }
+
+    public void setProject(String project) {
+        this.project = project;
+    }
+
+    @Override
+    protected List<Image> findAllGoogle(Compute client) throws Exception {
+        List<Image> images = new ArrayList<>();
+        ImageList imageList;
+        String nextPageToken = null;
+
+        do {
+            imageList = client.images().list(getProjectId()).setPageToken(nextPageToken).execute();
+            images.addAll(imageList.getItems().stream().filter(Objects::nonNull).collect(Collectors.toList()));
+            nextPageToken = imageList.getNextPageToken();
+        } while (nextPageToken != null);
+
+        return images;
+    }
+
+    @Override
+    protected List<Image> findGoogle(Compute client, Map<String, String> filters) throws Exception {
+        if (filters.containsKey("name")) {
+            return Collections.singletonList(client.images().get(
+                filters.containsKey("project") ? filters.get("project") : getProjectId(), filters.get("name"))
+                .execute());
+        } else if (filters.containsKey("family")) {
+            return Collections.singletonList(client.images().getFromFamily(
+                filters.containsKey("project") ? filters.get("project") : getProjectId(), filters.get("family"))
+                .execute());
+        } else {
+            List<Image> images = new ArrayList<>();
+            ImageList imageList;
+            String nextPageToken = null;
+
+            do {
+                imageList = client.images().list(filters.get("project")).setPageToken(nextPageToken).execute();
+                images.addAll(imageList.getItems().stream().filter(Objects::nonNull).collect(Collectors.toList()));
+                nextPageToken = imageList.getNextPageToken();
+            } while (nextPageToken != null);
+
+            return images;
```

Comment:
Would be a good to have a single return by declaring the list of images up top and returning it at the very ed, with the individual conditional scopes populating the list.